ui/src/components/shared/AutocompleteSearchBar.vue

```diff
@@ -0,0 +1,320 @@
+<template>
+  <div class="search-container">
+    <span class="p-input-icon-right search-group">
+      <i v-if="loading" class="pi pi-spin pi-spinner"></i>
+      <i v-else-if="speech" class="pi pi-microphone mic" :class="listening && 'listening'" @click="toggleListen"></i>
+      <InputText
+        id="autocomplete-search"
+        v-model="searchText"
+        :placeholder="searchPlaceholder"
+        @complete="debounceForSearch"
+        data-testid="search-input"
+        autofocus
+        v-on:keyup.enter="search()"
+        v-on:focus="showResultsOverlay"
+        v-on:blur="hideResultsOverlay"
+        @mouseover="showOverlay($event, selected?.iri)"
+        @mouseleave="hideOverlay($event)"
+      />
+    </span>
+    <OverlayPanel ref="resultsOP" :breakpoints="{ '960px': '75vw', '640px': '100vw' }" :style="{ width: '450px' }" appendTo="body">
+      <div v-if="loading" class="loading-container">
+        <ProgressSpinner />
+      </div>
+      <div v-else class="p-fluid results-container">
+        <span
+          >Showing {{ results?.entities?.length ? 1 : 0 }}-{{ results?.entities?.length ? results.entities.length : 0 }} of
+          {{ results?.count ? results.count : 0 }} results</span
+        >
+        <Listbox v-if="results?.entities" v-model="selectedLocal" :options="results.entities">
```

Comment:
Either limit results size to a number (e.g. first 100) or allow nested Virtualscroller https://primevue.org/listbox/#virtualscroll to prevent any performance issues